src/extractor/compressed_edge_container.cpp

```diff
@@ -250,16 +250,25 @@ CompressedEdgeContainer::GetBucketReference(const EdgeID edge_id) const
     return m_compressed_geometries.at(index);
 }
 
+bool CompressedEdgeContainer::ContainsCompressedGeometries(const EdgeID edge_id) const
+{
+    const auto &bucket = GetBucketReference(edge_id);
+    return bucket.size() > 1;
+}
+
 NodeID CompressedEdgeContainer::GetFirstEdgeTargetID(const EdgeID edge_id) const
 {
     const auto &bucket = GetBucketReference(edge_id);
-    BOOST_ASSERT(bucket.size() >= 2);
     return bucket.front().node_id;
 }
+NodeID CompressedEdgeContainer::GetLastEdgeTargetID(const EdgeID edge_id) const
+{
+    const auto &bucket = GetBucketReference(edge_id);
+    return bucket[bucket.size() - 1].node_id;
+}
 NodeID CompressedEdgeContainer::GetLastEdgeSourceID(const EdgeID edge_id) const
 {
     const auto &bucket = GetBucketReference(edge_id);
-    BOOST_ASSERT(bucket.size() >= 2);
```

Comment:
Don't remove but correct assertion for uncompressed case: `BOOST_ASSERT(bucket.size() >= 1);`